src/cryptonote_basic/account.cpp

```diff
@@ -60,7 +63,56 @@ DISABLE_VS_WARNINGS(4244 4345)
     m_device = &hwdev;
     MCDEBUG("device", "account_keys::set_device device type: "<<typeid(hwdev).name());
   }
-
+  //-----------------------------------------------------------------
+  static void derive_key(const crypto::chacha_key &base_key, crypto::chacha_key &key)
+  {
+    static_assert(sizeof(base_key) == sizeof(crypto::hash), "chacha key and hash should be the same size");
+    char data[sizeof(base_key) + 1];
+    memcpy(data, &base_key, sizeof(base_key));
+    data[sizeof(base_key)] = KEYS_ENCRYPTION_SALT;
+    crypto::generate_chacha_key(data, sizeof(data), key);
+    memwipe(data, sizeof(data));
+  }
+  //-----------------------------------------------------------------
+  static epee::wipeable_string get_key_stream(const crypto::chacha_key &base_key, const crypto::chacha_iv &iv, size_t bytes)
+  {
+    // derive a new key
+    crypto::chacha_key key;
+    derive_key(base_key, key);
+
+    // chacha
+    epee::wipeable_string buffer0(std::string(bytes, '\0'));
+    epee::wipeable_string buffer1 = buffer0;
+    crypto::chacha20(buffer0.data(), buffer0.size(), key, iv, buffer1.data());
```

Comment:
Got this error:
```
src/cryptonote_basic/account.cpp:87:5: error: no matching function for call to 'chacha20'
    crypto::chacha20(buffer0.data(), buffer0.size(), key, iv, buffer1.data());
    ^~~~~~~~~~~~~~~~
src/crypto/chacha.h:68:15: note: candidate function not viable: 5th argument ('const char *') would lose
      const qualifier
  inline void chacha20(const void* data, std::size_t length, const chacha_key& key, const chacha_iv& iv, char* cipher) {
              ^
```
because `wipeable_string::data()` returns `const char *`. But even with a `char *` overload as in #4116 the compiler seems to still stick with the `const` one, so cast `(char *)` seems necessary here.

Reply:
That's one change in a subsequent patch :/ I'll bring it forward.

Reply:
Does it work if you cherry pick 4116 first ? It builds here with CLANG without changing constness.